src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs

```diff
@@ -0,0 +1,589 @@
+                    {
+                        testInfos = FilterForSkippedRuntime(
+                            FilterForSkippedTargetFrameworkMonikers(
+                                DecorateWithSkipOnPlatform(testInfos, (int)filterAttribute.ConstructorArguments[1].Value!, options),
+                                (int)filterAttribute.ConstructorArguments[2].Value!),
+                            (int)filterAttribute.ConstructorArguments[3].Value!, options);
+                    }
+                    else
+                    {
+                        switch (filterAttribute.AttributeConstructor.Parameters[1].Type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat))
+                        {
+                            case "global::Xunit.TestPlatforms":
+                                testInfos = DecorateWithSkipOnPlatform(testInfos, (int)filterAttribute.ConstructorArguments[1].Value!, options);
+                                break;
+                            case "global::Xunit.TestRuntimes":
+                                testInfos = FilterForSkippedRuntime(testInfos, (int)filterAttribute.ConstructorArguments[1].Value!, options);
+                                break;
+                            case "global::Xunit.TargetFrameworkMonikers":
+                                testInfos = FilterForSkippedTargetFrameworkMonikers(testInfos, (int)filterAttribute.ConstructorArguments[1].Value!);
+                                break;
+                            default:
+                                break;
+                        }
+                    }
+                    break;
+                case "Xunit.SkipOnMonoAttribute":
+                    if (options.GlobalOptions.RuntimeFlavor() != "Mono")
+                    {
+                        // If we're building tests not for Mono, we can skip handling the specifics of the SkipOnMonoAttribute.
+                        testInfos = ImmutableArray<ITestInfo>.Empty;
+                        break;
+                    }
+                    testInfos = DecorateWithSkipOnPlatform(testInfos, (int)filterAttribute.ConstructorArguments[0].Value!, options);
+                    break;
+                case "Xunit.SkipOnPlatformAttribute":
+                    testInfos = DecorateWithSkipOnPlatform(testInfos, (int)filterAttribute.ConstructorArguments[0].Value!, options);
+                    break;
+                case "Xunit.PlatformSpecificAttribute":
+                    testInfos = DecorateWithSkipOnPlatform(testInfos, ~(int)filterAttribute.ConstructorArguments[0].Value!, options);
+                    break;
+                case "Xunit.SkipOnTargetFrameworkAttribute":
+                    testInfos = FilterForSkippedTargetFrameworkMonikers(testInfos, (int)filterAttribute.ConstructorArguments[0].Value!);
+                    break;
+                case "Xunit.SkipOnCoreClrAttribute":
+                    if (options.GlobalOptions.RuntimeFlavor() != "CoreCLR")
```

Comment:
Would it be a big problem to postpone the platform / runtime check to the actual runtime execution? We can do it in a future change but ideally we should be able to use the same managed build for Mono and CoreCLR (I don't believe we're doing that now, we have two different legs for that purpose but that's also cost that could be reduced).

Reply:
The only mechanism known today to detect runtime flavor at runtime easily is though reflection, which is very heavyweight for the src/tests test tree. I'm trying to keep the platform checks as cheap as possible so as to not make the "bring up" tests in the JIT significantly less useful, which is why I try to hard-code them at generation time when possible.

Reply:
I see. I think it's OK to leave as is for now, I can look into it as part of the new-style Helix publishing; in particular I believe we're setting up about two dozens of environment variables that get set on the Helix execution machine and these include target architecture and OS so I guess we could use these for the runtime checks. We can also add an arbitrary number of additional environment variables as we need to express any of the execution characteristics.